scarlet/operator.py

```diff
@@ -248,7 +249,56 @@ def prox_soft_symmetry(X, step, strength=1):
     return X
 
 
-def prox_uncentered_symmetry(X, step, center=None, strength=.5, use_soft=True, fill=None):
+def prox_kspace_symmetry(X, step, shift=None, padding=10):
+    """Symmetry in Fourier Space
+
+    This algorithm by Nate Lust uses the fact that throwing
+    away the imaginary part in Fourier space leaves a symmetric
+    soution in real space. So `X` is transformed to Fourier space,
+    shifted by the fractional amount `shift=(dy, dx)`,
+    the imaginary part is discarded, shited back to its original position,
+    then transformed back to real space.
+    """
+    # Record the morph shape
+    shape = X.shape
+    dy, dx = shift
+    padding = np.max(X.shape) + padding // 2
+    edges = ((padding, padding), (padding, padding))
+    corner = (padding, padding)
+    zeroMask = X <= 0
+    X[zeroMask] = 0
+    X = np.pad(X, edges, 'constant')
+
+    freq_x = np.fft.fftfreq(X.shape[1])
+    freq_y = np.fft.fftfreq(X.shape[0])
+
+    # Transform to k space
+    X_fft = np.fft.fftn(np.fft.ifftshift(X))
+
+    # Shift the signal to recenter it, negative because math is opposite from
+    # pixel direction
+    shifter = np.outer(np.exp(-1j*2*np.pi*freq_y*-(dy)),
+                       np.exp(-1j*2*np.pi*freq_x*-(dx)))
+    inv_shifter = np.outer(np.exp(-1j*2*np.pi*freq_y*(dy)),
+                           np.exp(-1j*2*np.pi*freq_x*(dx)))
+    result_fft = X_fft*shifter
+
+    # symmeterize
+    result_fft = result_fft.real
+
+    # Shift back
+    result_fft = result_fft*inv_shifter
+
+    # Transform to real space
+    result = np.fft.fftshift(np.fft.ifftn(result_fft))
+    # Return the unpadded transform
```

Comment:
I tested the code using `observation._centered` and it works the same, but that code has a few extra operations because the size of the padding is not known. Since it's only two lines of code I think it's best to leave as is, but if you feel strongly about duplicate code I can push a commit using `observation._centered`.

Reply:
I agree, it doesn't matter